File: Majority Element II.py
# https://leetcode.com/problems/majority-element-ii/

from collections import Counter

class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
                
        # Turn list to a dictionary
        nums = Counter(nums)
        
        # Iterate dictionary and save keys with matching maxVal
        res = list()
        for k, v in nums.items():
            
            # If value is greater than n/3, add it to list
            if v > (len(nums) / 3):
                res.append(k)
                
        return res

File: Majority Element II.java
// https://leetcode.com/problems/majority-element-ii/

class Solution {
    public List<Integer> majorityElement(int[] nums) {
        
        // Create a HashMap
        HashMap<Integer, Integer> counter = new HashMap<>();
        
        // Populate HashMap with elements from nums
        for (int elem: nums) {
            
            // Convert int to Integer
            Integer tmp = Integer.valueOf(elem);
            
            // Add Integer to HashMap
            counter.putIfAbsent(tmp, 0);
            counter.put(tmp, counter.get(tmp) + 1);
        }
        
        // Iterate HashMap
        ArrayList<Integer> res = new ArrayList<>();
        for (Map.Entry<Integer, Integer> set: counter.entrySet()) {
            
            if (set.getValue() > nums.length / 3) {
                res.add(set.getKey());
            }
        }
        return res;
    }
}
